In [12]:
library(ggplot2)
Honda_Accord<-read.csv("HondaAccord2.csv")

## Write down the complete second-order model for predicting a used car based on age and miles. 
$Y=B_0+B_1(X)+B_2(X^2)+B_3(X^3)+B_

In [13]:
SecondOrder_Model<- lm(Price~Age+Miles+I(Age^2)+I(Miles^2)+Age:Miles, data=Honda_Accord)
summary(SecondOrder_Model)
anova(SecondOrder_Model)


Call:
lm(formula = Price ~ Age + Miles + I(Age^2) + I(Miles^2) + Age:Miles, 
    data = Honda_Accord)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.8878 -1.0495  0.1057  0.8373  3.5583 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 25.272751   0.895285  28.229   <2e-16 ***
Age         -0.873077   0.902663  -0.967   0.3456    
Miles       -0.127789   0.065975  -1.937   0.0678 .  
I(Age^2)    -0.128606   0.264423  -0.486   0.6323    
I(Miles^2)   0.000171   0.001313   0.130   0.8978    
Age:Miles    0.013787   0.035637   0.387   0.7031    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.674 on 19 degrees of freedom
Multiple R-squared:  0.9207,	Adjusted R-squared:  0.8998 
F-statistic:  44.1 on 5 and 19 DF,  p-value: 8.307e-10


Df Sum Sq      Mean Sq     F value     Pr(>F)      
Age         1 595.2750199 595.2750199 212.5145873 9.084048e-12
Miles       1   2.0266907   2.0266907   0.7235334 4.055798e-01
I(Age^2)    1  12.5822018  12.5822018   4.4918758 4.745187e-02
I(Miles^2)  1   7.3342914   7.3342914   2.6183593 1.221146e-01
Age:Miles   1   0.4192596   0.4192596   0.1496767 7.031436e-01
Residuals  19  53.2209366   2.8011019          NA           NA

In [14]:
Miles_In_Quadratic<-lm(Price~Age+Miles+I(Age^2),data=Honda_Accord)
summary(Miles_In_Quadratic)
anova(Miles_In_Quadratic)


Call:
lm(formula = Price ~ Age + Miles + I(Age^2), data = Honda_Accord)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.0771 -0.6676 -0.2389  0.6101  3.3517 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 25.25692    0.89924  28.087  < 2e-16 ***
Age         -2.01650    0.41811  -4.823 9.12e-05 ***
Miles       -0.04177    0.02547  -1.640   0.1159    
I(Age^2)     0.08491    0.04079   2.082   0.0498 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.704 on 21 degrees of freedom
Multiple R-squared:  0.9091,	Adjusted R-squared:  0.8961 
F-statistic: 70.02 on 3 and 21 DF,  p-value: 4.204e-11


Df Sum Sq     Mean Sq    F value     Pr(>F)      
Age        1 595.275020 595.275020 205.0164899 2.628779e-12
Miles      1   2.026691   2.026691   0.6980051 4.128598e-01
I(Age^2)   1  12.582202  12.582202   4.3333901 4.979277e-02
Residuals 21  60.974488   2.903547          NA           NA

## Write down the prediction equation. 


$Price=25.272751 -0.873077(Age) -0.127789(Miles) -0.128606(Age^2) + 0.000171(Miles^2)+ 0.013787(Age x Miles)$

## Show how the details of a nested F test for just the second order terms (quadratic and interaction) that involves miles from this model. Include whatever output you need to get to the information to complete the test. Be sure to show all the usual details (hypothesis, test statistic, p value) and complete the test. In particular, show how the test statistic is computed by putting values into a formula. 

In [15]:
anova(Miles_In_Quadratic, SecondOrder_Model)

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 21     60.97449 NA       NA        NA        NA
2 19     53.22094  2 7.753551  1.384018 0.2747112

$H_0: B_4=B_5 = 0$

$H_a: B_4=B_5 \neq 0$



$F = \frac{\frac{SSE_(Reduced) - SSE_(Full)}{num. of predictors tested}}{SSE_(Full)/(n-k-1)}$

num. of predictors tested = 2

n=25

k=5

n-k-1: 25-5-1=19

$SSE_reduced = 60.97449$

$SSE_full= 53.22094$




In [16]:
SSE_reduced<-60.97449
SSE_Full<- 53.22094
num_pred<-2

F_stat<-((SSE_reduced-SSE_Full)/(2))/((SSE_Full)/(19))
F_stat


[1] 1.384018

Being that the p value (p=0.2747) is greater than alpha = 0.1, this implies that the second order terms (quadratic and ineteraction) that involves miles from this model are not strong predictors and thus, they're not helpful to his model. We could probably just use $Price=B_0+ B_1(Age)+ B_2(Miles)+B_3(Age^2) for predicting the price of used Honda Accords. 

## Wrap up: Based on the various models you have considered for predicting the price of a used car, which you would you recommend using in practice? Give some justification for your answer. 

I would not use the linear model because the higher the age, eventually there will be a negative price which is not possible unless the person is paying the seller to take the used car away.

The complete second order model isn't a good model because the predictors: miles, $miles^2$ and the interaction between Miles and Age are not strong predictors for the model because all of them have a p value greater than alpha =.1, which means they're not nededed in the model. 

Miles is not a strong predictor when included in the quadratic model as well.

I would recommend using the quadratic model because both predictors Age and $Age^2$ are important to the model and the price does not go to 0 or become negative.